python/GafferSceneUI/SceneInspector.py

```diff
@@ -1386,14 +1388,18 @@ def __parameters( self, target ) :
 
 			return shaders[-1].parameters  # only the last one is supported
 
-	def __init__( self, inspector, diffCreator = TextDiff, **kw ) :
+	def __init__( self, shaderName, shaderType, inspector, diffCreator = TextDiff, **kw ) :
```

Comment:
The shader name/type need to be computed on the fly rather than baked on construction like this. To see why, make a shader assignment, override it with a different shader downstream, view those parameters, and then disable the downstream node - the parameters will update dynamically in the inspector, but the name/type won't update. Similarly, when doing a diff, we need to display the diff for the name and type. We could do this using a TextDiff directly for each of name/type, and updating them in `update()`. Alternatively you could make a __NameAndTypeInspector and use a DiffColumn...